dev/package-examples/iptables-1.0.4/filebeat/config/input.yml

```diff
@@ -1,25 +1,32 @@
-{{ if eq .input "syslog" }}
-
+{{setEmptyVar input "syslog"}}
+{{ ifeq input "syslog" }}
 type: syslog
 protocol.udp:
-  host: "{{.syslog_host}}:{{.syslog_port}}"
+  host: "{{defVal syslog.host localhost}}:{{defVal syslog.post 9001}}"
 
-  {{ else if eq .input "file" }}
 
+{{ elseifeeq input "file" }}
 type: log
 paths:
-  {{ range $i, $path := .paths }}
-- {{$path}}
-  {{ end }}
+-
+  {{!-- If the paths variable is set, it will be used, otherwise falls back to the default config. --}}
```

Comment:
I'm realising now, that if we have the set for the default variable, we could skip this and instead set paths if not set yet.

Reply:
I'd prefer to have some more explicit way to define the defaults as we had with `manifest.yml`. Before, we had an only way to set variables in the fileset definition, and here I see three (`setEmptyVar`, `defVal` and this custom check).

Also we would still need some logic like the one offered previously by the `manifest.yml` file, where you could set defaults also per operating system, e.g. in the apache module:
```
  - name: paths
    default:
      - /var/log/apache2/access.log*
      - /var/log/apache2/other_vhosts_access.log*
      - /var/log/httpd/access_log*
    os.darwin:
      - /usr/local/var/log/apache2/access_log*
    os.windows:
      - "C:/tools/Apache/httpd-2.*/Apache24/logs/access.log*"
      - "C:/Program Files/Apache Software Foundation/Apache2.*/logs/access.log*"
```